tools/cold-store/src/cli.rs

```
@@ -384,3 +424,180 @@ impl PrepareHotCmd {
+            return Ok(());
+        }
+
+        let bytes = Self::read_state(store, hash.as_ref())
+            .with_context(|| format!("Failed to read raw bytes for hash {:?}", hash))?
+            .expect(format!("Failed to find raw bytes for hash {:?}", hash).as_str());
+        match near_store::RawTrieNodeWithSize::decode(&bytes) {
+            Ok(node) => match node.node {
+                near_store::RawTrieNode::Leaf(..) => {
+                    tracing::debug!(target: "check_trie", "Reached leaf node");
+                    return Ok(());
+                }
+                near_store::RawTrieNode::Branch(mut children, _) => {
+                    children.shuffle(&mut rand::thread_rng());
+                    for child in children.iter() {
+                        if let Some(child) = child {
+                            // Record in prune state that we are visiting a child node
+                            prune_state.down();
+                            // Visit a child node
+                            Self::check_trie(store, child, prune_state, prune_condition)?;
+                            // Record in prune state that we are returning from a child node
+                            prune_state.up();
+                        }
+                    }
+                }
+                near_store::RawTrieNode::Extension(_, child) => {
+                    // Record in prune state that we are visiting a child node
+                    prune_state.down();
+                    // Visit a child node
+                    Self::check_trie(store, &child, prune_state, prune_condition)?;
+                    // Record in prune state that we are returning from a child node
+                    prune_state.up();
+                }
+            },
+            Err(_) => {
+                tracing::debug!(target: "check_trie", "Reached leaf node");
+                return Ok(());
```

This seems wrong, should you return an error from here?